src/mammos_mumag/mesh.py

```diff
@@ -1,8 +1,97 @@
 """Mesh functions."""
 
 import pathlib
+import shutil
 
-_MESH_DIR = pathlib.Path(__file__).parent / "mesh"
+import requests
+import yaml
 
-CUBE_20_nm = _MESH_DIR / "CUBE_20_nm.fly"
-"""Cube with 20 nanometer sides."""
+with open(pathlib.Path(__file__).parent / "mesh" / "README.yaml") as f:
+    DATA = yaml.safe_load(f)
+
+
+def find_mesh(mesh_name: str | None = None) -> list[str]:
+    """Find available meshes matchin given name.
+
+    Args:
+        mesh_name: Desired mesh name. If None, returns all available meshes.
+
+    Returns:
+        List of matches with given name. Empty list if no matches are found.
+    """
+    if mesh_name is None:
+        return list(DATA["meshes"].keys())
+    else:
+        return [n for n in DATA["meshes"] if mesh_name in n]
+
+
+class Mesh:
+    """Mesh class."""
+
+    def __init__(self, mesh_name: str | pathlib.Path):
+        """Initialize Mesh with either mesh_name or path for local meshes."""
+        matches = (
+            find_mesh(mesh_name) if not isinstance(mesh_name, pathlib.Path) else []
+        )
+        if len(matches) > 1:
+            raise ValueError("Mesh name unambiguous. More than one match found.")
```

Comment:
```suggestion
            raise ValueError("Mesh name ambiguous. More than one match found.")
```
We could also display the matches.